### PR TITLE
flow logs default update & required_provider 1.3.0

### PR DESCRIPTION
**Pull request by pablo19sc**
- Removed "experiments" block and update required_provider to >= 1.3.0 in all the provider definition (main providers.tf, submodules, and examples).
- Updated VPC Flow Logs variable definition to make use of the new "optional" functionality for object type.
- Minor changes in "flow logs" submodule (typo in one variable), in README.md (to add the proper link to 3.0 changes), and examples.

As some changes were done to the code, I attach here the result of different tests - the first one is the example of the flow logs piece (which is the main change in this PR).

flowlog_test
transit_gateway_test
secondary_cidr_test
natgw_route_test

### CODE REVIEW

File: .header.md [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 This module can be used to deploy a pragmatic VPC with various subnets types in # AZs. Common deployment examples can be found in [examples/](https://github.com/aws-ia/terraform-aws-vpc/tree/main/examples). Subnet CIDRs can be explicitly set via list of string argument `cidrs` or set via a number `netmask` argument.
 
-_Note: For information regarding the 2.0 upgrade see our [upgrade guide](https://github.com/aws-ia/terraform-aws-vpc/blob/main/UPGRADE-GUIDE-2.0.md)._
+_Note: For information regarding the 3.0 upgrade see our [upgrade guide](https://github.com/aws-ia/terraform-aws-vpc/blob/main/UPGRADE-GUIDE-3.0.md)._
```
Comment (drewmullen) [inline .header.md:5]: THANK YOU FOR DOING THIS!

File: main.tf [hunk 2]
```diff
@@ -257,7 +257,8 @@ module "flow_logs" {
 
   name = var.name
   # see defaults.tf for local definition
-  flow_log_defintion = local.flow_logs_definition
-  vpc_id             = local.vpc.id
-  tags               = module.tags.tags_aws
+  # flow_log_definition = local.flow_logs_definition
```
Comment (drewmullen) [inline main.tf:260]: please remove these 2 unnecessary comments

File: variables.tf [hunk 3]
```diff
@@ -183,16 +183,16 @@ variable "vpc_flow_logs" {
...
-      per_hour_partition         = optional(bool)
+      file_format                = optional(string, "plain-text")
+      hive_compatible_partitions = optional(bool, false)
+      per_hour_partition         = optional(bool, false)
     }))
   })
 
   default = {
-    log_destination_type = "none"
+    log_destination_type = "cloud-watch-logs"
```
Comment (drewmullen) [inline variables.tf:195]: By default we probably do not want to build flow logs since they can get quite expensive

File: examples/ipam/main.tf [hunk 4]
```diff
@@ -1,6 +1,5 @@
 module "vpc" {
-  source  = "aws-ia/vpc/aws"
```
Comment (drewmullen) [inline examples/ipam/main.tf:2]: for each example, can you flip it back from `source = "../.." to the module source in tf regstiry

**Review by drewmullen — CHANGES_REQUESTED** [review]

**Comment by pablo19sc** [comment]
`terraform validate` in all the examples (showing validation passes)

Screenshot 2022-10-20 at 12 15 43
Screenshot 2022-10-20 at 12 14 52
Screenshot 2022-10-20 at 12 10 10
Screenshot 2022-10-20 at 12 09 24

**Pull request merged**